frontend/app/src/composables/settings/api-keys/external/index.ts

```diff
@@ -83,11 +113,23 @@ export const useExternalApiKeys = createSharedComposable(
         [key]: message,
       });
     };
-    const save = async ({ name, apiKey }: { name: string; apiKey: string }) => {
+    const save = async (payload: ExternalServiceKey) => {
+      const { name } = payload;
       resetStatus(name);
       try {
         set(loading, true);
-        set(keys, await setExternalServices([{ name, apiKey: apiKey.trim() }]));
+
+        const trimmedPayload: Writeable<ExternalServiceKey> = payload;
+        if ('apiKey' in trimmedPayload) {
+          trimmedPayload.apiKey = trimmedPayload.apiKey.trim();
+        }
+        else {
+          trimmedPayload.username = trimmedPayload.username.trim();
+          trimmedPayload.password = trimmedPayload.password.trim();
```

Comment:
Wouldn't v-model.trim work here and save use some re-assigning?

Reply:
ah yes, updating...